broker-core/src/main/java/io/zeebe/broker/clustering/base/topology/TopologyManagerImpl.java

```diff
@@ -170,8 +170,10 @@ public void onEvent(SocketAddress sender, DirectBuffer payload) {
             readSocketAddress(offset, payloadCopy, replicationApi);
 
             final NodeInfo newMember = new NodeInfo(clientApi, managementApi, replicationApi);
-            topology.addMember(newMember);
-            notifyMemberAdded(newMember);
+            final boolean memberAdded = topology.addMember(newMember);
+            if (memberAdded) {
```

Comment:
just for better understanding, why was this change needed?

Reply:
I thought that it make no sense to notify the listeners if the member already exist, but maybe this can not happen.

Reply:
it's fine, was sure if this is now a requirement that it never happens twice, but if it's just a optimization than it's fine